src/main/java/onlinebookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,36 @@
+package onlinebookstore.repository.impl;
+
+import java.util.List;
+import onlinebookstore.model.Book;
+import onlinebookstore.repository.BookRepository;
+import org.hibernate.SessionFactory;
+import org.hibernate.query.Query;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```

Comment:
1. @Autowired is optional on constructors
2. you can use Lombok`s `@RequiredArgsConstructor`  and get rid of this constructor at all - it works well with Spring constructor injection